File: scripts/algorithms/A/Arranging Coins/Arranging Coins.java
// Runtime: 5 ms (Top 44.14%) | Memory: 40.7 MB (Top 83.77%)
class Solution {
    public int arrangeCoins(int n) {
        long s =0; long e = n;
        while (s <= e) {
         long mid = s +(e-s)/2;
            long coin = mid *( mid +1)/2;
            if(coin > n){
                 e = mid -1;
            } else if (coin < n){
                s = mid +1;
            } else return (int) mid;
        }
        return (int)e;
    }
}

File: scripts/algorithms/A/Arranging Coins/Arranging Coins.py
# Runtime: 6683 ms (Top 5.01%) | Memory: 13.8 MB (Top 57.45%)
class Solution:
    def arrangeCoins(self, n: int) -> int:
        for i in range(1,2**31):
            val=i*(i+1)//2
            if val>n:
                a=i
                break
            elif val==n:
                return i
        return a-1